### PR TITLE
Escape schema for getProcedures and getProcedureColumns in SQLServerDatabaseMetaData

### PR DESCRIPTION
**Pull request by tkyc**

### CODE REVIEW

**Review by barryw-mssql — DISMISSED** [review]

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDatabaseMetaData.java [hunk 1]
```diff
@@ -1425,7 +1425,7 @@ public java.sql.ResultSet getProcedureColumns(String catalog, String schema, Str
         // proc name supports escaping
         proc = escapeIDName(proc);
         arguments[0] = proc;
-        arguments[1] = schema;
+        arguments[1] = escapeIDName(schema);
```
Comment (lilgreenbird) [inline src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDatabaseMetaData.java:1428]: the comments need updating. They call out proc and col specifically is it implying that the others don't support escaping? This is also making me wondering about catalog does this have the same issue?

Comment (Jeffery-Wasty) [reply src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDatabaseMetaData.java:1428]: My suggestion for lines 1426 - 1432:

```suggestion
        // proc, schema, and col names support escaping
        arguments[0] = escapeIDName(proc);
        arguments[1] = escapeIDName(schema);
        arguments[2] = catalog;
        arguments[3] = escapeIDName(col);
```

File: src/test/java/com/microsoft/sqlserver/jdbc/TestUtils.java [hunk 2]
```diff
@@ -419,6 +419,15 @@ public static void dropTableIfExists(String tableName, java.sql.Statement stmt)
         dropObjectIfExists(tableName, "U", stmt);
     }
 
+    public static void dropTableWithSchemaIfExists(String tableNameWithSchema, java.sql.Statement stmt) throws SQLException {
+        stmt.execute("IF OBJECT_ID('" + tableNameWithSchema + "', 'U') IS NOT NULL DROP TABLE " + tableNameWithSchema + ";");
+    }
+
+
+    public static void dropProcedureWithSchemaIfExists(String procedureWithSchema, java.sql.Statement stmt) throws SQLException {
+        stmt.execute("IF EXISTS (SELECT * FROM sys.objects WHERE object_id = OBJECT_ID(N'" + procedureWithSchema + "') AND type in (N'P', N'PC')) DROP PROCEDURE " + procedureWithSchema + ";");
```
Comment (Jeffery-Wasty) [inline src/test/java/com/microsoft/sqlserver/jdbc/TestUtils.java:428]: Nitpick: linelength, please split into 2 lines.
```suggestion
        stmt.execute("IF EXISTS (SELECT * FROM sys.objects WHERE object_id = OBJECT_ID(N'" + procedureWithSchema 
        + "') AND type in (N'P', N'PC')) DROP PROCEDURE " + procedureWithSchema + ";");
```

File: src/test/java/com/microsoft/sqlserver/jdbc/databasemetadata/DatabaseMetaDataTest.java [hunk 3]
```diff
@@ -1000,6 +1024,9 @@ public static void setupTable() throws Exception {
                     + " ([col_1] int NOT NULL, [col%2] varchar(200), [col[3] decimal(15,2))");
             stmt.execute("CREATE FUNCTION " + AbstractSQLGenerator.escapeIdentifier(functionName)
                     + " (@p1 INT, @p2 INT) RETURNS INT AS BEGIN DECLARE @result INT; SET @result = @p1 + @p2; RETURN @result; END");
+            stmt.execute("CREATE SCHEMA " + schema);
+            stmt.execute("CREATE TABLE " + tableNameWithSchema + " (id UNIQUEIDENTIFIER, name NVARCHAR(400));");
+            stmt.execute("CREATE PROCEDURE " + sprocWithSchema + "(@id UNIQUEIDENTIFIER, @name VARCHAR(400)) AS BEGIN SET TRANSACTION ISOLATION LEVEL SERIALIZABLE BEGIN TRANSACTION UPDATE " + tableNameWithSchema + " SET name = @name WHERE id = @id COMMIT END");
```
Comment (Jeffery-Wasty) [inline src/test/java/com/microsoft/sqlserver/jdbc/databasemetadata/DatabaseMetaDataTest.java:1029]: Nitpick: linelength, please split into 3 lines
```suggestion
            stmt.execute("CREATE PROCEDURE " + sprocWithSchema + "(@id UNIQUEIDENTIFIER, @name VARCHAR(400)) AS" 
            + " BEGIN SET TRANSACTION ISOLATION LEVEL SERIALIZABLE BEGIN TRANSACTION UPDATE " 
            + tableNameWithSchema + " SET name = @name WHERE id = @id COMMIT END");
```

**Review by Jeffery-Wasty — DISMISSED** [review]

**Review by Jeffery-Wasty — APPROVED** [review]

**Review by barryw-mssql — APPROVED** [review]

**Review by lilgreenbird — APPROVED** [review]

**Pull request merged**